asv2/protocol.md

```diff
@@ -0,0 +1,270 @@
+# Attestations Protocol
+
+The goals of this attestation protocol are to increase adoption of identity mappings, and encourage network effects of shared cross-application identity attestations. To this end, we have made it possible for anyone to permissionlessly become an issuer of attestations.
+
+  - [User Flows](#user-flows)
+    - [Register Alice’s phone number](#register-alices-phone-number)
+    - [Look up Alice’s phone number](#look-up-alices-phone-number)
+    - [Send money to user on a different wallet](#send-money-to-user-on-a-different-wallet)
+  - [Interacting with the protocol](#interacting-with-the-protocol)
+    - [Issuer Signers](#issuer-signers)
+    - [Registration](#registration)
+      - [Registering identifiers as the issuer](#registering-identifiers-as-the-issuer)
+      - [Registering identifiers with signer](#registering-identifiers-with-signer)
+    - [Lookups](#lookups)
+  - [Smart Contract Addresses](#smart-contract-addresses)
+    - [Mainnet](#mainnet)
+    - [Alfajores](#alfajores)
+
+## User Flows
+
+<details>
+<summary><b>Stakeholders</b></summary>
+
+| Stakeholder | Description |
+|-----------|--------|
+| user | User that owns the off-chain identifier and account
+| issuer | Anyone who is verifying identifiers and creating on-chain attestations, this will most likely be a wallet  
+| Oblivious Decentralised Identity Service (ODIS) | An API that produces secrets used for obfuscating the identifiers, more details [here](privacy.md) |
+| `FederatedAttestations.sol` | On-chain registry of attestation mappings between identifiers and addresses |
+| `OdisPayments.sol` |  Smart contract used to pay for [ODIS quota](privacy.md#rate-limit) |
+
+</details>
+
+<!-- Here is a visual overview:
+
+<img width="1200" alt="image" src="https://user-images.githubusercontent.com/46296830/201716282-39e1b1b9-7a88-4e2c-8607-417ddcec2443.png"> -->
+
+### Register phone number
+
+The user provides their phone number to the application, who is acting as an issuer. The application verifies that the user has ownership of their phone number. Then the application queries ODIS for the obfuscated identifier of the phone number. Using the obfuscated identifier, the application, as an issuer, registers the on-chain mapping between the identifier and the user's account address.
+
+<img width="1200" alt="image" src="https://user-images.githubusercontent.com/46296830/201714875-c73d8417-0e0c-47b4-9b41-8529689f0607.png">
+
+### Look up phone number 
+
+The wallet queries ODIS for the obfuscated identifier of the phone number. Using the obfuscated identifier, the wallet looks up the on-chain registry to see which account is mapped to that identifier.
+
+<img width="1200" alt="image" src="https://user-images.githubusercontent.com/46296830/201715097-124a8461-2a45-4a1f-ab2a-1781300befb0.png">
+
+### Send money to user on a different wallet
```

Comment:
It would be good if this section had a short written summary like the other sections